cpp/oneapi/dal/backend/primitives/stat/cov_dpc.cpp

```diff
@@ -57,6 +57,132 @@ inline void validate_input(const sycl::queue& q,
     ONEDAL_ASSERT(is_known_usm(q, tmp.get_mutable_data()));
 }
 
+template <typename Float>
+inline void validate_input_cov(const sycl::queue& q,
+                               const ndview<Float, 2>& data,
+                               const ndview<Float, 1>& sums,
+                               const ndview<Float, 2>& cov,
+                               const ndview<Float, 1>& means,
+                               const ndview<Float, 1>& vars) {
+    ONEDAL_ASSERT(data.has_data());
+    ONEDAL_ASSERT(sums.has_data());
+    ONEDAL_ASSERT(cov.has_mutable_data());
+    ONEDAL_ASSERT(means.has_mutable_data());
+    ONEDAL_ASSERT(vars.has_mutable_data());
+    ONEDAL_ASSERT(cov.get_dimension(0) == cov.get_dimension(1), "Covariance matrix must be square");
+    ONEDAL_ASSERT(cov.get_dimension(0) == data.get_dimension(1),
+                  "Dimensions of covariance matrix must match feature count");
+    ONEDAL_ASSERT(sums.get_dimension(0) == data.get_dimension(1),
+                  "Element count of sums must match feature count");
+    ONEDAL_ASSERT(vars.get_dimension(0) == data.get_dimension(1),
+                  "Element count of vars must match feature count");
+    ONEDAL_ASSERT(means.get_dimension(0) == data.get_dimension(1),
+                  "Element count of means must match feature count");
+    ONEDAL_ASSERT(is_known_usm(q, sums.get_data()));
+    ONEDAL_ASSERT(is_known_usm(q, data.get_data()));
+    ONEDAL_ASSERT(is_known_usm(q, cov.get_mutable_data()));
+    ONEDAL_ASSERT(is_known_usm(q, means.get_mutable_data()));
+    ONEDAL_ASSERT(is_known_usm(q, vars.get_mutable_data()));
+}
+
+template <typename Float>
+inline sycl::event compute_means(sycl::queue& q,
+                                 const ndview<Float, 2>& data,
+                                 const ndview<Float, 1>& sums,
+                                 ndview<Float, 1>& means,
+                                 const event_vector& deps) {
+    const std::int64_t column_count = data.get_dimension(1);
+    const std::int64_t row_count = data.get_dimension(0);
+
+    const Float inv_n = Float(1.0 / double(row_count));
+
+    const Float* sums_ptr = sums.get_data();
+    Float* means_ptr = means.get_mutable_data();
+
+    return q.submit([&](sycl::handler& cgh) {
+        const auto range = ::oneapi::dal::backend::make_multiple_nd_range_1d(
+            column_count,
+            ::oneapi::dal::backend::device_max_wg_size(q));
```

Comment:
I'm not sure if it is correct. I propose one of the following solution:

Reply:
1) Let compiler decide which WG size would be optimal in this case and just use trivial `sycl::range(column_count)` and `sycl::id<1>`.
2) Use not `column_count`, but smth like that:
```suggestion
        const auto wg = propose_wg_size(q);
        const auto bcount = (column_count / wg) + bool(column_count % wg);
        const auto fcount = wg * bcount;
        const auto range = make_multiple_nd_range_1d(fcount, wg);
```